R/union.R

```diff
@@ -41,6 +48,23 @@ new_union <- function(...) {
   out
 }
 
+#' @export
+`|.S7_class` <- function(e1, e2) {
+  new_union(e1, e2)
+}
+#' @export
+`|.S7_union` <- `|.S7_class`
+#' @export
+`|.S7_base_class` <- `|.S7_class`
```

Comment:
Being able to define methods for `class_character` vs `"character"` is pretty nice. i.e. `"integer" | "character"` wouldn't work the same

Reply:
Yeah, I think I put off implementing this because I was worried about introducing ambiguities with `|`, but it lucky there aren't any any more 😄